src/main/java/com/example/macro/user/service/impl/UserServiceImp.java

```diff
@@ -0,0 +1,67 @@
+package com.example.macro.user.service.impl;
+
+import com.example.macro.user.document.User;
+import com.example.macro.user.exception.DuplicateUser;
+import com.example.macro.user.exception.UserNotFoundException;
+import com.example.macro.user.repository.UserRepository;
+import com.example.macro.user.service.UserService;
+import com.example.macro.user.utils.EmailValidation;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class UserServiceImp implements UserService {
+
+    private final BCryptPasswordEncoder passwordEncoder;
+
+    private final UserRepository userRepository;
+
+    public UserServiceImp(BCryptPasswordEncoder passwordEncoder, UserRepository userRepository) {
+        this.passwordEncoder = passwordEncoder;
+        this.userRepository = userRepository;
+    }
+
+    @Override
+    public User createUser(User user) {
+
+        if (existingUser(user.getEmail())) {
+            throw new DuplicateUser("Record with email " + user.getUserName() + " Already Exists");
+        }
+
+        if (EmailValidation.isEmailValid(user.getEmail())) {
+            user.setPassword(passwordEncoder.encode(user.getPassword()));
+            return userRepository.save(user);
+        }
+
+        return new User();
```

Comment:
Does this return an empty user object if neither the DuplicateUserException is true or the save is successful? 

Should we specify something else if both above conditions fail? Returning an empty object seems like a strange choice.